### ISSUE TITLE
Fix S2275: Rule should not report on multiple ":" in formatting methods

### ISSUE THREAD

**Issue by Evangelink** [SonarSource/sonar-csharp#222]
The following examples:
Example 1:
`Console.WriteLine("{0:HH:mm:ss} - {1}", DateTime.Now, text);`

Example 2:
`string.Format("{0} - ({1:dd-MM-yyyy HH:mm:ss}).mht", email.Headers.Subject.Limitar(100, false), DateTime.Now);`

Example 3:
`string.Format("{0:###,###,###}", Math.Round(pi.QtCadenciaDiaria.GetValueOrDefault(0)).ToString());`

will report issues where they shouldn't. The problem lays in the analysis of the format string item when we do the split using ":" and ",".

**Comment by jairbubbles** [comment]
Same issue on my side with:
```csharp
string.Format("@{0:yyyy/MM/dd:HH:mm:ss}", DateTime.Now)
```
Not sure if / is also problematic or not.

**Comment by Evangelink** [comment]
Hi @jairbubbles! You can safely ignore this case. As you may have noticed we are currently working on the fix for this bug and it will be available in the next release/version.

**Comment by jairbubbles** [comment]
I was just adding more info in case it was a different case.  
   
In the future you should consider putting a low severity on new rules. S2275 was a critical one and we had some build failing after adding updating the plugin (we're using Sonar Way quality profile)

Anyway, looking forward to the new release!
